search/scorer/scorer_term.go

```diff
@@ -159,7 +159,9 @@ func (s *TermQueryScorer) Score(ctx *search.SearchContext, termMatch *index.Term
 
 	rv := ctx.DocumentMatchPool.Get()
 	rv.IndexInternalID = append(rv.IndexInternalID, termMatch.ID...)
-	rv.Score = score
+	if s.options.Score != "none" {
```

Comment:
hmmm, thinking a string compare in the "hot loop" of scoring every term match might have a perf impact?

Reply:
fair, let me add a bool flag to the scorer to avoid this string compare.